Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp

```diff
@@ -88,84 +93,112 @@ namespace Multiplayer
+            m_byteStream.Read(sizeof(uint32_t), &hintSize);
+            assetHint.resize(hintSize);
+            m_byteStream.Read(hintSize, assetHint.data());
+            size_t assetSize = m_byteStream.GetCurPos();
+
+            // Load spawnable from stream without loading any asset references
+            AzFramework::Spawnable* spawnable = AZ::Utils::LoadObjectFromStream<AzFramework::Spawnable>(
+                m_byteStream, nullptr, AZ::ObjectStream::FilterDescriptor(AZ::Data::AssetFilterNoAssetLoading));
+            if (!spawnable)
             {
-                AZ::Data::AssetId assetId;
-                uint32_t hintSize;
-                AZStd::string assetHint;
-                m_byteStream.Read(sizeof(AZ::Data::AssetId), reinterpret_cast<void*>(&assetId));
-                m_byteStream.Read(sizeof(uint32_t), reinterpret_cast<void*>(&hintSize));
-                assetHint.resize(hintSize);
-                m_byteStream.Read(hintSize, assetHint.data());
-
-                size_t assetSize = m_byteStream.GetCurPos();
-                AZ::Data::AssetData* assetDatum = AZ::Utils::LoadObjectFromStream<AZ::Data::AssetData>(m_byteStream, nullptr);
-                if (!assetDatum)
-                {
-                    AZLOG_ERROR("EditorServerLevelData packet contains no asset data. Asset: %s", assetHint.c_str())
-                    return false;
-                }
-                assetSize = m_byteStream.GetCurPos() - assetSize;
-                AZ::Data::Asset<AZ::Data::AssetData> asset = AZ::Data::Asset<AZ::Data::AssetData>(assetId, assetDatum, AZ::Data::AssetLoadBehavior::NoLoad);
-                asset.SetHint(assetHint);
-
-                AZ::Data::AssetInfo assetInfo;
-                assetInfo.m_assetId = asset.GetId();
-                assetInfo.m_assetType = asset.GetType();
-                assetInfo.m_relativePath = asset.GetHint();
-                assetInfo.m_sizeBytes = assetSize;
-
-                // Register Asset to AssetManager
-                AZ::Data::AssetManager::Instance().AssignAssetData(asset);
-                AZ::Data::AssetCatalogRequestBus::Broadcast(&AZ::Data::AssetCatalogRequests::RegisterAsset, asset.GetId(), assetInfo);
-
-                assetData.push_back(asset);
+                AZLOG_ERROR("EditorServerLevelData packet contains no asset data. Asset: %s", assetHint.c_str())
+                return false;
             }
 
-            // Now that we've deserialized, clear the byte stream
-            m_byteStream.Seek(0, AZ::IO::GenericStream::SeekMode::ST_SEEK_BEGIN);
-            m_byteStream.Truncate();
+            // We only care about Root.spawnable and Root.network.spawnable
+            if (!assetHint.starts_with(AzFramework::Spawnable::DefaultMainSpawnableName))
+            {
+                AZ_Assert(false, "Editor sent the server more than just the root (level) spawnable. Ensure the editor code only sends Root.");
```

Comment:
NP: If you're not doing anything else here you could just AZ_Assert the condition instead of doing an if an AZ_Assert(false).

Reply:
Good catch, fixed!